addons/base_import_module/models/ir_module.py

```diff
@@ -124,7 +124,7 @@ def _import_module(self, module, path, force=False):
 
         # Look for existing assets
         existing_assets = IrAsset.search([('name', 'in', [vals['name'] for vals in assets_vals])])
-        existing_assets = existing_assets.mapped(lambda r: (r.name, r))
+        existing_assets = {existing_asset.name: existing_asset for existing_asset in existing_assets}
```

Comment:
Assigning a variable with different things makes the code hard to read, IMO.
I suggest to rewrite the two lines as follows
```python
existing_assets = {
    asset.name: asset
    for asset in IrAsset.search([('name', 'in', [vals['name'] for vals in assets_vals])])
}
```